Subscriber/DoctrineToGatewayEventSubscriber.php

```diff
@@ -0,0 +1,324 @@
+        $event = new ActionEvent('commongateway.object.post.create', ['object' => $object]);
+        $this->eventDispatcher->dispatch($event, 'commongateway.object.post.create');
+    }
+
+    /**
+     * Updated object in database.
+     *
+     * @param LifecycleEventArgs $args
+     *
+     * @return void
+     */
+    public function postUpdate(LifecycleEventArgs $args): void
+    {
+        $object = $args->getObject();
+
+        // if this subscriber only applies to certain entity types,
+        if (!$object instanceof ObjectEntity) {
+            return;
+        }
+
+        // Write the log
+        $this->logger->info(
+            'Updated object in database',
+            [
+                'object'    => $object->getId(),
+                'entity'    => $object->getEntity()->getId()
+            ]
+        );
+
+        // Throw the event
+        $event = new ActionEvent('commongateway.object.post.update', ['object' => $object]);
+        $this->eventDispatcher->dispatch($event, 'commongateway.object.post.update');
+    }
+
+    /**
+     * Read object from database.
+     *
+     * @param LifecycleEventArgs $args
+     *
+     * @return void
+     */
+    public function postLoad(LifecycleEventArgs $args): void
+    {
+        $object = $args->getObject();
+
+        // if this subscriber only applies to certain entity types,
+        if (!$object instanceof ObjectEntity) {
+            return;
+        }
+
+        // Write the log
+        $this->logger->info(
+            'Read object from database',
+            [
+                'object'    => $object->getId(),
+                'entity'    => $object->getEntity()->getId()
+            ]
+        );
+
+        // Throw the event
+        $event = new ActionEvent('commongateway.object.post.read', ['object' => $object]);
+        $this->eventDispatcher->dispatch($event, 'commongateway.object.post.read');
+    }
+
+    /**
+     * Flushing entity manager.
+     *
+     * @param LifecycleEventArgs $args
+     *
+     * @return void
+     */
+    public function preFlush(PreFlushEventArgs $args): void
+    {
+        // Write the log
+        $this->logger->info(
+            'Flushing entity manager',
+            [
+            ]
+        );
+
+        // Throw the event
+        $event = new ActionEvent('commongateway.object.pre.flush', []);
+        $this->eventDispatcher->dispatch($event, 'commongateway.object.pre.flush');
+    }
+
+    /**
+     * Flushed entity manager.
+     *
+     * @param LifecycleEventArgs $args
+     *
+     * @return void
+     */
+    public function postFlush(PostFlushEventArgs $args): void
+    {
+
+        // Write the log
+        $this->logger->info(
+            'Flushed entity manager',
+            [
+            ]
+        );
+
+        // Throw the event
+        $event = new ActionEvent('commongateway.object.post.flush', []);
+        $this->eventDispatcher->dispatch($event, 'commongateway.object.post.flush');
+    }
+}
```

Comment:
I'm not sure what will be the effect of passing the full object to the event manager, it might be that you end up with unusable objects in the action handlers.